lib/tcp/tcp_packet.ml

```diff
@@ -73,7 +73,6 @@ module Marshal = struct
   let unsafe_fill ~pseudoheader ~payload t buf options_len =
     let data_off = (sizeof_tcp / 4) + (options_len / 4) in
     let buf = Cstruct.sub buf 0 (data_off * 4) in
-    Cstruct.memset buf 0; (* autopad *)
```

Comment:
Thanks @yomimono, @avsm, the test looks good to me too.  Was just discussing with @MagnusS - another test he is thinking of adding is to examine a small packet capture to see if behaves as expected - eg, in this case check that the SYN pkts have MSS options.